usr/share/rear/output/RAWDISK/Linux-i386/270_create_grub2_efi_bootloader.sh

```diff
@@ -19,7 +19,15 @@ fi
 # (3) Grub 2 EFI components must exist
 # Since openSUSE Leap 15.1 things were moved from /usr/lib/grub2/ to /usr/share/grub2/
 # cf. https://github.com/rear/rear/issues/2338#issuecomment-594432946
-[[ -d /usr/lib/grub/x86_64-efi ]] || [[ -d /usr/lib/grub2/x86_64-efi ]] || [[ -d /usr/share/grub2/x86_64-efi ]] || return 0
+local efi_modules_directory
+local dir
+for dir in /usr/lib/grub/x86_64-efi /usr/lib/grub2/x86_64-efi /usr/share/grub2/x86_64-efi; do
+    if [[ -d "$dir" ]]; then
+        efi_modules_directory="$dir"
+        break
+    fi
+done
+[[ -z "$efi_modules_directory" ]] && return 0
 
```

Comment:
Only FYI if you like to simplify the code in the following way:
I think you could omit the helper variable `dir` because you coud do
```
# Use first found GRUB2 UEFI modules directory or skip the script if none exists:
local efi_modules_directory
for efi_modules_directory in /usr/lib/grub/x86_64-efi /usr/lib/grub2/x86_64-efi /usr/share/grub2/x86_64-efi ; do
    test -d "$efi_modules_directory && break
done
test -d "$efi_modules_directory" || return 0
```

Reply:
Yes surely that would work, too. In this case I'd like to stick with `dir` because it's a bit more explicit and thus I think it's easier to grasp. Your's is shorter but looks a bit more tricky to me.

Reply:
@OliverO2 
I fully agree with you.

Cf. https://github.com/rear/rear/wiki/Coding-Style (excerpts):
```
Introduce intermediate variables with meaningful names
to tell what is going on.
...
Better very simple code than oversophisticated
(possibly fragile) constructs. 
```

By the way and only for the fun of it:
I found an even more "elegant" (a.k.a. "tricky") way to do all inside the `for` loop:
```
# Use first found GRUB2 UEFI modules directory or skip the script if none was found:
local efi_modules_directory
for efi_modules_directory in /usr/lib/grub/x86_64-efi /usr/lib/grub2/x86_64-efi /usr/share/grub2/x86_64-efi not_found ; do
    test -d "$efi_modules_directory" && break
    test 'not_found' = "$efi_modules_directory" && return 0
done
```